assets/js/blocks/product-collection/inspector-controls/price-range-control/PriceTextField.tsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	// @ts-expect-error Using experimental features
+	__experimentalNumberControl as NumberControl,
+	// @ts-expect-error Using experimental features
+	__experimentalInputControlPrefixWrapper as InputControlPrefixWrapper,
+} from '@wordpress/components';
+
+interface PriceTextFieldProps {
+	value: number;
+	onChange: ( min: number ) => void;
+	label?: string;
+}
+
+const PriceTextField: React.FC< PriceTextFieldProps > = ( {
+	value,
+	onChange,
+	label,
+} ) => {
+	return (
+		<NumberControl
+			value={ value }
+			onChange={ ( min: string ) => {
+				onChange( Number( min ) );
```

Comment:
Sam as above: Component is used for both: min and max so I think it should be agnostic:

```suggestion
			onChange={ ( value: string ) => {
				onChange( Number( value ) );
```